content/learn/book/migration-guides/0.9-0.10/_index.md

```diff
@@ -0,0 +1,1043 @@
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">UI</div>
+</div>
+
+The `alignment` field of `Text` now only affects the text’s internal alignment.
+
+__Change `TextAlignment` to TextAlignment` which is now an enum. Replace:__
+
+- `TextAlignment::TOP_LEFT`, `TextAlignment::CENTER_LEFT`, `TextAlignment::BOTTOM_LEFT` with `TextAlignment::Left`
+- `TextAlignment::TOP_CENTER`, `TextAlignment::CENTER_LEFT`, `TextAlignment::BOTTOM_CENTER` with `TextAlignment::Center`
+- `TextAlignment::TOP_RIGHT`, `TextAlignment::CENTER_RIGHT`, `TextAlignment::BOTTOM_RIGHT` with `TextAlignment::Right`
+
+__Changes for `Text2dBundle`__
+
+`Text2dBundle` has a new field ‘text_anchor’ that takes an `Anchor` component that controls its position relative to its transform.
+
+### [Remove `QueuedText`](https://github.com/bevyengine/bevy/pull/7414)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">UI</div>
+</div>
+
+`QueuedText` was never meant to be user facing. If you relied on it, please make an issue.
+
+### [change the default `width` and `height` of `Size` to `Val::Auto`](https://github.com/bevyengine/bevy/pull/7475)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">UI</div>
+</div>
+
+The default values for `Size` `width` and `height` have been changed from `Val::Undefined` to `Val::Auto`.
+It’s unlikely to cause any issues with existing code.
+
+### [Fix the `Size` helper functions using the wrong default value and improve the UI examples](https://github.com/bevyengine/bevy/pull/7626)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">UI</div>
+</div>
+
+The `Size::width` constructor function now sets the `height` to `Val::Auto` instead of `Val::Undefined`.
+The `Size::height` constructor function now sets the `width` to `Val::Auto` instead of `Val::Undefined`.
+
+### [The `size` field of `CalculatedSize` should not be a `Size`](https://github.com/bevyengine/bevy/pull/7641)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">UI</div>
+</div>
+
+- The size field of `CalculatedSize` has been changed to a `Vec2`.
+
+### [Windows as Entities](https://github.com/bevyengine/bevy/pull/5589)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">Windowing</div>
+</div>
+
+- Replace `WindowDescriptor` with `Window`.
+  - Change `width` and `height` fields in a `WindowResolution`, either by doing
+
+```rust
+WindowResolution::new(width, height) // Explicitly
+// or using From<_> for tuples for convenience
+(1920., 1080.).into()
+```
+
+- Replace any `WindowCommand` code to just modify the `Window`’s fields directly  and creating/closing windows is now by spawning/despawning an entity with a `Window` component like so:
+
+```rust
+let window = commands.spawn(Window { ... }).id(); // open window
+commands.entity(window).despawn(); // close window
+```

```

Comment:
This is missing a paragraph explaining how to replace usage of `Res<Windows>` here is an example:

----

Now that windows are components on entity, you should use `Query` instead of `Res` to access windows

**Before:**
```rust
fn count_pixels(windows: Res<Windows>) {
  let Some(primary) = windows.get_primary() else { return; };
  println!("{}", primary.width() * primary.height());
}
```
**After:**
```rust
use bevy::window::PrimaryWindow;
fn count_pixels(primary_query: Query<&Window, With<PrimaryWindow>>) {
  let Ok(primary) = primary_query.get_single() else { return; };
  println!("{}", primary.width() * primary.height());
}
```

Reply:
```suggestion
### [Windows as Entities](https://github.com/bevyengine/bevy/pull/5589)

<div class="migration-guide-area-tags">
    <div class="migration-guide-area-tag">Windowing</div>
</div>

- Replace `WindowDescriptor` with `Window`.
  - Change `width` and `height` fields in a `WindowResolution`, either by doing

```rust
WindowResolution::new(width, height) // Explicitly
// or using From<_> for tuples for convenience
(1920., 1080.).into()
```

- Replace any `WindowCommand` code to just modify the `Window`’s fields directly  and creating/closing windows is now by spawning/despawning an entity with a `Window` component like so:

```rust
let window = commands.spawn(Window { ... }).id(); // open window
commands.entity(window).despawn(); // close window
```

Now that windows are components on entity, you should use Query instead of Res to access windows

```rust
fn count_pixels(windows: Res<Windows>) { let Some(primary) = windows.get_primary() else { return; }; println!("{}", primary.width() * primary.height()); }
```

```rust
use bevy::window::PrimaryWindow; fn count_pixels(primary_query: Query<&Window, With<PrimaryWindow>>) { let Ok(primary) = primary_query.get_single() else { return; }; println!("{}", primary.width() * primary.height()); }
```

```